include/glow/Partitioner/Partitioner.h

```diff
@@ -143,28 +138,24 @@ class Partitioner {
 
   /// Based on partitionConfig_ passed into Partitioner, do the user-defined
   /// partition.
-  llvm::Error PartitionFromConfig();
-
-  /// Decompose each function in a module. Now we support partitioning a module
-  /// among different type of devices. \p cctx is used during optimization of
-  /// the Function. \returns whether there was an error encountered.
-  llvm::Error Partition(CompilationContext &cctx);
+  llvm::Expected<DAGListTy>
+  partitionFromConfig(PartitionConfig &partitionConfig);
 
   /// This partition approach is used in Glow Quantization Profiling flow. The
   /// backendBasedPartition is applied first in case there are heterogeneous
   /// backends. Then each sub-function will be compiled and run in CPU backend
   /// for profiling.
-  llvm::Error QuantizationProfilingPartition(CompilationContext &cctx,
-                                             Function *F,
-                                             std::vector<Backend *> backends);
+  llvm::Expected<DAGListTy>
+  quantizationProfilingPartition(CompilationContext &cctx);
```

Comment:
why all these special methods are part of the public interface?

Reply:
Discussed with @artemrakhov before. It allows users to call those method directly. For example,  user-defined partitioner  can be called directly without setting params, e.g "optimized_" and "saturateHost_".  @artemrakhov could you please also help to double check the interface? Thanks!

Reply:
Yes, this is the right direction! It is very useful to be able to call just one method.

But Roman wanted to bring a different question to your attention. Can we create more Partitioner-based (inherited) classes? I.e. we could further introduce `UserDefinedPartitioner`, `SaturatingPartitioner`, and so on. Whatever makes sense. That way, `Partitioner` class itself won't implement any "special" partitions. It will be a very general class.

Reply:
> Yes, this is the right direction! It is very useful to be able to call just one method.
> 
> But Roman wanted to bring a different question to your attention. Can we create more Partitioner-based (inherited) classes? I.e. we could further introduce `UserDefinedPartitioner`, `SaturatingPartitioner`, and so on. Whatever makes sense. That way, `Partitioner` class itself won't implement any "special" partitions. It will be a very general class.

We now have "class PartitionerBase" and new partitioner class can be derived from that. The reason we have current partition flows in Partitioner because they shares lots of common data types, internal members.  We would like to keep "class PartitionerBase" stateless.

Reply:
We can do further inheritance. For example: PartitionerBase -> Partitioner -> UserDefinedPartitioner